exp/api/v1alpha3/awsfargateprofile_types.go

```diff
@@ -0,0 +1,162 @@
+	// Labels specifies which pod labels this selector should match
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// Namespace specifies which namespace this selector should match
+	Namespace string `json:"namespace,omitempty"`
+}
+
+// FargateProfileStatus defines the observed state of FargateProfile
+type FargateProfileStatus struct {
+	// Ready denotes that the FargateProfile has been created
+	// +kubebuilder:default=false
+	Ready bool `json:"ready"`
+
+	// FailureReason will be set in the event that there is a terminal problem
+	// reconciling the MachinePool and will contain a succinct value suitable
+	// for machine interpretation.
+	//
+	// This field should not be set for transitive errors that a controller
+	// faces that are expected to be fixed automatically over
+	// time (like service outages), but instead indicate that something is
+	// fundamentally wrong with the Machine's spec or the configuration of
+	// the controller, and that manual intervention is required. Examples
+	// of terminal errors would be invalid combinations of settings in the
+	// spec, values that are unsupported by the controller, or the
+	// responsible controller itself being critically misconfigured.
+	//
+	// Any transient errors that occur during the reconciliation of MachinePools
+	// can be added as events to the MachinePool object and/or logged in the
+	// controller's output.
+	// +optional
+	FailureReason *errors.MachineStatusError `json:"failureReason,omitempty"`
+
+	// FailureMessage will be set in the event that there is a terminal problem
+	// reconciling the MachinePool and will contain a more verbose string suitable
+	// for logging and human consumption.
+	//
+	// This field should not be set for transitive errors that a controller
+	// faces that are expected to be fixed automatically over
+	// time (like service outages), but instead indicate that something is
+	// fundamentally wrong with the MachinePool's spec or the configuration of
+	// the controller, and that manual intervention is required. Examples
+	// of terminal errors would be invalid combinations of settings in the
+	// spec, values that are unsupported by the controller, or the
+	// responsible controller itself being critically misconfigured.
+	//
+	// Any transient errors that occur during the reconciliation of MachinePools
+	// can be added as events to the MachinePool object and/or logged in the
+	// controller's output.
+	// +optional
+	FailureMessage *string `json:"failureMessage,omitempty"`
+
+	// Conditions defines current service state of the managed machine pool
+	// +optional
+	Conditions clusterv1.Conditions `json:"conditions,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=awsfargateprofiles,scope=Namespaced,categories=cluster-api
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Ready",type="string",JSONPath=".status.ready",description="AWSFargateProfile ready status"
```

Comment:
Perhaps some additional items here like profilename, failure message. What do you think?

Reply:
added profileName and failureReason